File: 2021-August/04/java/src/member/controller/MemberManager.java
package member.controller;

import member.model.vo.Member;

public class MemberManager {

	private static int pMember = 0;
	private static final int MAX_MEMBER = 40;

	private Member[] member = new Member[MAX_MEMBER];

	public void insertMember(Member member) {
		this.member[pMember++] = member;
	}

	public void printData() {
		System.out.println("---------------------------<<회원정보>>---------------------------");
		System.out.printf("%s\t\t%s\t\t%s\t\t%s%n", "이름", "등급", "포인트", "이자포인트");
		System.out.println("-----------------------------------------------------------------");

		for (int i = 0; i < pMember; i++) {
			System.out.println(this.formatData(this.member[i]));
		}
	}

	private String formatData(Member member) {
		return String.format("%s\t\t%s\t\t%d\t\t%.2f", member.getName(), member.getGrade(), member.getPoint(),
				member.getInterestPoint());
	}

	public void printBuyInfo(int price) {
		for (Member member : member) {
			if (member == null)
				break;
			System.out.printf("%s\t등급회원 %s은 %d원 상품을 %d원에 구매합니다.%n", member.getGrade(), member.getName(), price,
					member.buy(price));

		}
	}

}

File: 2021-July/30/src/member/controller/MemberManager.kt
package member.controller

import member.model.vo.Silver

class MemberManager {

    companion object {
        private var pSilver: Int = 0
        private val MAX_MEMBER: Int = 10
    }

    private val silver: Array<Silver?> = arrayOfNulls<Silver>(MAX_MEMBER)



    fun silverInsert(silver: Silver?): Unit {
        this.silver[pSilver++] = silver
    }

    fun printData(): Unit {
        println("---------------------------<<회원정보>>---------------------------");
        println("이름\t\t등급\t\t포인트\t\t이자포인트")
        println("-----------------------------------------------------------------");

            for(i in 0 until pSilver)
            {
                println(this.formatData(this.silver[i]))
            }
    }

    private fun formatData(s: Silver?): String {

        // if(s != null) s.toUpperCase()
        // else null

        // s?.toUpperCase()

        // if (s == null) return ""
        // else 
            return s?.name + "\t\t" + s?.grade + "\t\t" + s?.point + "\t\t" + s?.getInterest()
    }
    
}